pronotepy/ent.py

```diff
@@ -715,42 +611,85 @@ def l_normandie(username:str,password:str):
     """
-    
+
     # URL required
     teleservicesUrl = "https://teleservices.education.gouv.fr/mdp/Shibboleth.sso/Login?SAMLDS=1&target=ss:mem:0cbabc441187d789fdc65845af6ea345affc926e159a67ea87c5f4cae48c5962&authnContextClassRef=urn:oasis:names:tc:SAML:2.0:ac:classes:PasswordProtectedTransport urn:oasis:names:tc:SAML:2.0:ac:classes:TimeSyncToken&entityID=https://educonnect.education.gouv.fr/idp"
     educonnectUrl = "https://educonnect.education.gouv.fr/idp/profile/SAML2/Unsolicited/SSO?providerId=https://ent.l-educdenormandie.fr/auth/saml/metadata/idp.xml&service=https://0142131r.index-education.net/pronote/mobile.eleve.html"
     educdenormandieUrl = "https://ent.l-educdenormandie.fr/auth/saml/acs"
-    
+
     # Headers
     headers = {
     'connection': 'keep-alive',
     'User-Agent': 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:73.0) Gecko/20100101 Firefox/73.0'}
-    
-    
+
     payload = {
         'j_username': username,
         'j_password': password,
         "_eventId_proceed": ""}
 
-    with requests.Session() as session:
+    session = requests.Session()
         
-        url = session.get(teleservicesUrl)
+    url = session.get(teleservicesUrl)
 
-        response = session.post(url.url,headers=headers,data=payload)
+    response = session.post(url.url,headers=headers,data=payload)
+
+    response = session.get(educonnectUrl)
+    soup = BeautifulSoup(response.text, 'html.parser')
+    payload2 = {
+        "SAMLResponse": soup.find("input", {"name": "SAMLResponse"})["value"]
+    }
+
+    log.debug(f'[ENT Educ de Normandie] Logging in with {username}')
+    response = session.post(educdenormandieUrl,headers=headers,data=payload2)
 
-        response = session.get(educonnectUrl)
-        soup = BeautifulSoup(response.text, 'html.parser')
-        payload2 = {
-            "SAMLResponse": soup.find("input", {"name": "SAMLResponse"})["value"]
-        }
-        
-        log.debug('[ENT Educ de Normandie] Logging in with ' + username)
-        response = session.post(educdenormandieUrl,headers=headers,data=payload2)
-        
     return requests.utils.cookiejar_from_dict(requests.utils.dict_from_cookiejar(session.cookies))
+
+def ent(url: str, username: str, password: str) -> requests.cookies.RequestsCookieJar:
+    """
+    ENT for given url (if supported and no specific function already exists) or can pass url as {'essonne', 'hauts-de-france', 'ile-de-france'}
+
+    Parameters
+    ----------
+    url : str
+        url of the ENT or {'essonne', 'hauts-de-france', 'ile-de-france'}
+    username : str
+        username
+    password : str
+        password
+
+    Returns
+    -------
+    cookies : cookies
+        returns the ent session cookies
+    """
+
+    ent = {
+        "essonne": "https://www.moncollege-ent.essonne.fr/auth/login",
+        "hauts-de-france": "https://enthdf.fr/auth/login",
+        "ile-de-france": "https://ent.iledefrance.fr/auth/login?callback=https%3A%2F%2Fent.iledefrance.fr%2F"
+    }
```

Comment:
This can be replaced by wrappers. They are more intuitive and don't break the API:
```py
def ent_essonne(username: str, password: str):
    return ent("https://www.moncollege-ent.essonne.fr/auth/login", username, password)
```